### DISCUSSION TITLE
2 types of capital costs in storage units (€/MW of power capacity installed and €/MWh of energy storage capacity installed)

### DISCUSSION THREAD

**Discussion by AntonisCY97** [PyPSA/PyPSA#1086] [Q&A] [answered]
Hello, I have a question regarding the costing of storage units in Pypsa. For example, let's say I want to install a set of batteries to provide flexibility and stop curtailments in an energy system. This storage unit will have 2 kind of capital costs.. A capital cost for it's installed power (e.g. if it can charge/discharge at a rate of 200 kW and it has a cost of 100 €/kW, it will cost 20000 €) and a capital cost for its capacity, more specifically, the amount of kWh it can store (e.g. if it can store up to 1000 kWh and this costs 75 €/kWh, it will cost 75000 €). So, after all this battery will have a capital cost of € 90000.

How can I implement this in Pypsa? As I can see, the capital cost refers only to currency/MW...

I hope you get it and help me :)

**Comment by gincrement** [comment, marked as answer]
This can be done by spliting the storage from the dis/charger equipment. 
As such you model a link with costs per MW, a storage with costs per MWh, and another link for discharging without costs but making sure the size of both links (dis/charging) have to be the same via a custom constraint.
